modules/external-dns/variables.tf

```diff
@@ -10,9 +10,9 @@ variable "chart_version" {
 }
 
 variable "values" {
-  description = "Additional values to pass to the Helm chart"
-  type        = map(any)
-  default     = {}
+  description = "Additional raw yaml values to pass to the Helm chart"
+  type        = list(string)
+  default     = []
```

Comment:
This absolutely needs to be `map(any)` because it is the arbitrary helm values. Then these are added to the list. This allows the implementation of a model to add any values you need.

Reply:
@kferrone the type is inherited from the underlying `helm_release` resource: https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release#values

That way the meaning of `values` remains the same in the module interface as it does in the `helm_release` interface, rather than rewriting it.

While looking at that again I think we may need a `concat()` call to join [these lists](https://github.com/duplocloud/terraform-kubernetes-addons/blob/50573c72cc307f027e8b4b3476a8e38f63db3868/modules/external-dns/main.tf#L21-L22) though otherwise it'll be a list nested in another list. @duplocloud-matt what do you think?

Reply:
Kelly and I discussed and sadly we gave up on elegantly passing values in some kind of easy to use format.  Regarding the concat, helm provider accepts values as a list so you can pass in multiple values files.  It will merge them, so we don't really have to concatenate them unless we have a compelling reason.  https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release#values says
"Values will be merged, in order, as Helm does with multiple -f options."  From a usability perspective, there's the values we set in the module that you populate with required variables.  Then the user can pass in values, presumably in the form of 
```
values = [
    "${file("values.yaml")}"
  ]
```
 That will merge our module values and their values, with theirs taking precedence, then anything the user passes in with 'sets' will override any of the values..  I believe 99/100 times the user will choose to pass in sets because of the friendly format.  We should also support set_lists at some point (https://github.com/hashicorp/terraform-provider-helm/issues/92) but I am a little stuck on this MR with Albedo so I'm just plowing ahead.